### PR TITLE
Error cleanup

### PR DESCRIPTION
**Pull request by dtracers**
Added more types of errors for better granularity in metrics

### CODE REVIEW

File: backend/blueprints/spa_api/errors/errors.py [hunk 1]
```diff
@@ -17,11 +17,23 @@ def to_dict(self):
         }
 
 
+class NotLoggedIn(CalculatedError):
+    status_code = 403
```
Comment (twobackfromtheend) [inline backend/blueprints/spa_api/errors/errors.py]: I think this should be a 401 or 404.

File: backend/blueprints/spa_api/service_layers/replay/tag.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def rename(current_name: str, new_name: str, session=None) -> 'Tag':
         # Check if name already exists
         try:
             TagWrapper.get_tag_by_name(session, get_current_user_id(), new_name)
-            raise CalculatedError(409, f"Tag with name {new_name} already exists.")
+            raise TagError(409, f"Tag with name {new_name} already exists.")
```
Comment (twobackfromtheend) [inline backend/blueprints/spa_api/service_layers/replay/tag.py]: TagError definition above has a 500 error while this (the error's only usage as far as I can tell) uses a 409?

File: backend/blueprints/spa_api/service_layers/player/play_style.py [hunk 3]
```diff
@@ -40,7 +40,7 @@ def create_from_id(cls, id_: str, raw=False, rank=None, replay_ids=None, playlis
                                                                     rank=rank, replay_ids=replay_ids,
                                                                     playlist=playlist, win=win)
         except NoResultFound:
-            raise CalculatedError(404, f"User has no replays for the selected playlist: {playlist}")
+            raise NoReplaysForPlaylist(f"User has no replays for the selected playlist: {playlist}")
```
Comment (twobackfromtheend) [inline backend/blueprints/spa_api/service_layers/player/play_style.py]: message is duplicated here and in the init above. Would give "User has no replays for the selected playlist: User has no replays for the selected playlist: {playlist}"

File: backend/blueprints/spa_api/service_layers/replay/tag.py (hunk 2 above)
Comment (dtracers) [reply backend/blueprints/spa_api/service_layers/replay/tag.py]: It is also a super class for tagnotfound

File: backend/blueprints/spa_api/errors/errors.py (hunk 1 above)
Comment (dtracers) [reply backend/blueprints/spa_api/errors/errors.py]: General internet consensus from what I saw said 403 is good for a not logged in request but I can change to 401

**Pull request merged**